raven/src/main/java/com/getsentry/raven/RavenContext.java

```diff
@@ -0,0 +1,117 @@
+package com.getsentry.raven;
+
+import com.getsentry.raven.event.Breadcrumb;
+import com.getsentry.raven.util.CircularFifoQueue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * RavenContext is used to hold {@link ThreadLocal} context data (such as
+ * {@link Breadcrumb}s) so that data may be collected in different parts
+ * of an application and then sent together when (e.g.) an exception occurs.
+ */
+public class RavenContext implements AutoCloseable {
+
+    /**
+     * Thread local set of active context objects. Note that an {@link IdentityHashMap}
+     * is used instead of a Set because there is no identity-set in the Java
+     * standard library.
+     */
+    private static ThreadLocal<IdentityHashMap<RavenContext, RavenContext>> activeContexts =
+        new ThreadLocal<IdentityHashMap<RavenContext, RavenContext>>() {
+            @Override
+            protected IdentityHashMap<RavenContext, RavenContext> initialValue() {
+                return new IdentityHashMap<>();
+            }
+    };
```

Comment:
Yeah sorry, I took the code as it existed here without adding the explanation. I'll add a comment explaining it now.